### ISSUE TITLE
CTCLoss value issue

### ISSUE THREAD

**Issue by floatn** [pytorch/pytorch#69016]
I'm working on CTC loss, and I have a feeling the loss value is too low in most cases.
For example, 

```python
import torch
import torch.nn

T = 10
C = 5
N = 1
S = 4

input = torch.full(size=(T, N, C), fill_value = 1/C).log_softmax(2).detach().requires_grad_()
target = torch.randint(low=1, high=C, size=(N, S), dtype=torch.long)
input_lengths = torch.full(size=(N,), fill_value=T, dtype=torch.long)
target_lengths = torch.randint(low=S, high=S + 1, size=(N,), dtype=torch.long)
ctc_loss = torch.nn.CTCLoss()
loss = ctc_loss(input, target, input_lengths, target_lengths)
```
, which in general case gives out loss = 2.02
But, I consider it should be at least higher than -log((3/C)^T) ~ 5.11
What do I miss?

**Comment by t-vi** [comment]
In [the documentation](https://pytorch.org/docs/stable/generated/torch.nn.CTCLoss.html) it says that with `reduction="mean"` the mean is taken over the target length. Is `reduction="sum"` what you are looking for?

**Comment by floatn** [comment]
@t-vi 
Yes, I think it is. Thank you!
